stan/math/rev/fun/squared_distance.hpp

```diff
@@ -130,27 +115,26 @@ class squared_distance_vd_vari : public vari {
 };
 }  // namespace internal
 
-template <int R1, int C1, int R2, int C2>
-inline var squared_distance(const Eigen::Matrix<var, R1, C1>& v1,
-                            const Eigen::Matrix<var, R2, C2>& v2) {
-  check_vector("squared_distance", "v1", v1);
-  check_vector("squared_distance", "v2", v2);
+template <
+    typename EigVecVar1, typename EigVecVar2,
+    require_all_eigen_vector_vt<is_var, EigVecVar1, EigVecVar2>* = nullptr>
+inline var squared_distance(const EigVecVar1& v1, const EigVecVar2& v2) {
   check_matching_sizes("squared_distance", "v1", v1, "v2", v2);
   return var(new internal::squared_distance_vv_vari(v1, v2));
```

Comment:
[optional]

return type can let this be 
```suggestion
  return {new internal::squared_distance_vv_vari(v1, v2)};
```

Reply:
fixed